### PR TITLE
feat(core): introduce `status_kind` in the `Context`

### PR DESCRIPTION
**Pull request by nerodono**
When user has special logic for, say, termination (i.e. waiting for some other actors to finish), it has to keep track of status as the internal state:
```rust
struct MyState {
    // .. some useful data about actor's internal state ..
    terminating: bool, // <-- flag that actor should do clean-up actions before termination
}
```

This can cause bugs that are difficult to catch (forgetting about the terminating flag when updating the state). Instead, we can provide some API to read actor's status, so user is not obligated with keeping `status`-related information in the internal state.

### CODE REVIEW

File: elfo-core/src/context.rs [hunk 1]
```diff
@@ -105,6 +106,14 @@ impl<C, K> Context<C, K> {
         ward!(self.actor.as_ref().and_then(|o| o.as_actor())).set_status(status);
     }
 
+    /// Gets the actor's status kind.
```
Comment (loyd) [inline elfo-core/src/context.rs:109]: example (with `is_terminating()`)

File: elfo-core/src/context.rs [hunk 2]
```diff
@@ -105,6 +106,14 @@ impl<C, K> Context<C, K> {
...
+    pub fn status_kind(&self) -> Option<ActorStatusKind> {
```
Comment (loyd) [inline elfo-core/src/context.rs:110]: Non-optional (panic in case of a pruned ctx)

File: elfo-core/src/actor.rs [hunk 3]
```diff
@@ -306,6 +306,10 @@ impl Actor {
         self.control.write().restart_policy = policy;
     }
 
+    pub(crate) fn status_kind(&self) -> ActorStatusKind {
```
Comment (loyd) [inline elfo-core/src/actor.rs:309]: I think we should implement `is_terminating()` and so on on `ActorStatusKind`

**Comment by loyd** [comment]
Add a record into `CHANGELOG.md` please with reference to the current PR

**Comment by nerodono** [comment]
Resolved everything at [73db5e3](https://github.com/elfo-rs/elfo/pull/133/commits/73db5e37a77a53c06ca19f12004854dcee7e007a) (needs careful review)
Changedlog at [2f7bbc8](https://github.com/elfo-rs/elfo/pull/133/commits/2f7bbc86a21dc0f0031a01aebd2fe369fbc0f0a1)

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased] - ReleaseDate
 ### Added
+- core: `Context::status_kind` API, now actors can read `ActorStatusKind` from the context.
+- core: `is_*` methods on `ActorStatusKind` for each variant.
```
Comment (loyd) [inline CHANGELOG.md:12]: Links (`... ([#133]).`)

File: elfo-core/src/actor.rs [hunk 5]
```diff
@@ -105,6 +110,32 @@ pub enum ActorStatusKind {
     Failed,
 }
 
+impl ActorStatusKind {
+    pub const fn is_normal(&self) -> bool {
```
Comment (loyd) [inline elfo-core/src/actor.rs:114]: Mark as `#[inline]` to allow inlining inter-crates

File: elfo-core/src/atomic_status_kind.rs [hunk 6]
```diff
@@ -0,0 +1,29 @@
+use std::mem;
+use std::sync::atomic::{self, AtomicU8};
+
```
Comment (loyd) [inline elfo-core/src/atomic_status_kind.rs:3]: I think it's better to move `ActorStatusKind` (and maybe `ActorStatus`) here and name this file "actor_status.rs".

`permissions.rs`, for instance, contain both `Permissions` and `AtomicPermissions`.

File: elfo-core/src/context.rs [hunk 7]
```diff
@@ -105,6 +106,30 @@ impl<C, K> Context<C, K> {
...
+    /// # fn exec(ctx: elfo::Context) {
+    /// // if actor is terminating.
+    /// assert!(ctx.status_kind().is_terminating());
+    /// // if actor is alarming.
+    /// assert!(ctx.status_kind().is_alarming());
+    /// // and so on...
+    /// # }
+    /// ```
+    /// # Panics
+    /// - Panics when called on pruned context.
```
Comment (loyd) [inline elfo-core/src/context.rs:123]: 1. Add an empty line before this section
2. What's the point to use a list here (-)?

File: elfo-core/src/lib.rs [hunk 8]
```diff
@@ -76,6 +76,8 @@ mod supervisor;
 mod telemetry;
 mod thread;
```
Comment (loyd) [inline elfo-core/src/lib.rs:78]: Why another mod's group (empty line between `mod`s)?

File: elfo-core/src/actor.rs [hunk 9]
```diff
@@ -105,6 +110,32 @@ pub enum ActorStatusKind {
...
+    }
+
+    pub const fn is_alarming(&self) -> bool {
+        matches!(self, Self::Alarming)
+    }
+
+    pub const fn is_failed(&self) -> bool {
+        matches!(self, Self::Failed)
+    }
+}
```
Comment (loyd) [inline elfo-core/src/actor.rs:137]: It seems that `Actor::is_terminating()` and `Actor::is_initializing()` can be replaced now by using `ctx.status_kind().is_*`

Comment (nerodono) [reply elfo-core/src/actor.rs]: Sure, did that in [1070425](https://github.com/elfo-rs/elfo/pull/133/commits/1070425102a5c11c57e2f0617259ec6dbfb6c18c)

File: elfo-core/src/lib.rs (hunk 8 above)
Comment (nerodono) [reply elfo-core/src/lib.rs]: Initially I thought they're kinda unrelated, fixed in [1070425](https://github.com/elfo-rs/elfo/pull/133/commits/1070425102a5c11c57e2f0617259ec6dbfb6c18c)

File: elfo-core/src/context.rs (hunk 7 above)
Comment (nerodono) [reply elfo-core/src/context.rs]: Done in [1070425](https://github.com/elfo-rs/elfo/pull/133/commits/1070425102a5c11c57e2f0617259ec6dbfb6c18c).
list is just a silly habit.

File: elfo-core/src/atomic_status_kind.rs (hunk 6 above)
Comment (nerodono) [reply elfo-core/src/atomic_status_kind.rs]: Done in [1070425](https://github.com/elfo-rs/elfo/pull/133/commits/1070425102a5c11c57e2f0617259ec6dbfb6c18c).

File: CHANGELOG.md (hunk 4 above)
Comment (nerodono) [reply CHANGELOG.md]: Done by [23befa9](https://github.com/elfo-rs/elfo/pull/133/commits/23befa9f672390f8413a0d68c306e5af67921631)

**Comment by nerodono** [comment]
One moment, seems like I need to fix up more

File: elfo-core/src/lib.rs [hunk 10]
```diff
@@ -10,8 +10,9 @@ extern crate elfo_utils;
 extern crate self as elfo_core;
 
 // TODO: revise this list
+pub use crate::actor_status::{ActorStatus, ActorStatusKind};
 pub use crate::{
```
Comment (loyd) [inline elfo-core/src/lib.rs:14]: merge reexports to make it consistent

**Review by loyd — APPROVED** [review]

**Pull request merged**